lib/connectors/memory.js

```diff
@@ -352,29 +352,61 @@ Memory.prototype.fromDb = function(model, data) {
   if (!data) return null;
   data = deserialize(data);
   var props = this._models[model].properties;
+
   for (var key in data) {
     var val = data[key];
-    if (val === undefined || val === null) {
-      continue;
-    }
+
     if (props[key]) {
-      switch (props[key].type.name) {
-        case 'Date':
-          val = new Date(val.toString().replace(/GMT.*$/, 'GMT'));
-          break;
-        case 'Boolean':
-          val = Boolean(val);
-          break;
-        case 'Number':
-          val = Number(val);
-          break;
-      }
+      val = this._castPropertyValue(key, val, props);
     }
+
     data[key] = val;
   }
+
   return data;
 };
 
+Memory.prototype._castPropertyValue = function(prop, val, props) {
+  if (val === undefined || val === null) {
+    return val;
+  }
+
+  if (Array.isArray(val)) {
+    var self = this;
+    return val.map(function(val) {
+      return self._castPropertyValue(prop, val, props);
+    });
+  }
+
+  var isArray = Array.isArray(props[prop].type);
```

Comment:
I also recommend to store `props[prop]` in a new variable, so that you don't have to repeat this snippet again and again in the code below.

```js
var propType = props[prop]; // or even better: var propertyType = ...
if (!propType) return val;
// ...
```